server/routes/calendar.js

```diff
@@ -1,63 +1,64 @@
-
-        const shiftNotifications = await notificationService.getShiftNotifications(req.user.username)
-
-        const apiShiftsResponse = await calendarService.getCalendarData(
-          userAuthenticationDetails[0].ApiUrl,
-          req.params.date,
-          req.user.token,
-        )
-
-        const apiOvertimeShiftsResponse = await calendarOvertimeService.getCalendarOvertimeData(
-          userAuthenticationDetails[0].ApiUrl,
-          req.params.date,
-          req.user.token,
-        )
-
-        const results = utilities.processOvertimeShifts(apiShiftsResponse, apiOvertimeShiftsResponse)
-
-        res.render('pages/calendar', {
-          shiftNotifications,
-          tab: 'Calendar',
-          startDate: req.params.date,
-          data: results,
-          uid: req.user.username,
-          employeeName: req.user.employeeName,
-          csrfToken: res.locals.csrfToken,
-          hmppsAuthMFAUser: req.hmppsAuthMFAUser,
-          authUrl: req.authUrl,
-        })
-      } catch (error) {
-        serviceUnavailable(req, res)
-      }
-    }),
-  )
-
-  // eslint-disable-next-line func-names
-  router.get('*', function (req, res) {
-    logger.info('Catch and redirect to current month view')
-    res.redirect(`/calendar/${utilities.getStartMonth()}`)
+  res.render('pages/index', {
+    authError: false,
+    csrfToken: res.locals.csrfToken,
   })
-
-  return router
 }
+router.get(
+  '/:date',
+  asyncMiddleware(async (req, res) => {
+    logger.info('GET calendar view')
+    const { calendarService, calendarOvertimeService, notificationService, userAuthenticationService } = req.app.get(
+      'DataServices',
+    )
+
+    try {
+      const userAuthenticationDetails = await userAuthenticationService.getUserAuthenticationDetails(req.user.username)
+
+      const shiftNotifications = await notificationService.getShiftNotifications(req.user.username)
+
+      const apiShiftsResponse = await calendarService.getCalendarData(
+        userAuthenticationDetails[0].ApiUrl,
+        req.params.date,
+        req.user.token,
+      )
+
+      const apiOvertimeShiftsResponse = await calendarOvertimeService.getCalendarOvertimeData(
+        userAuthenticationDetails[0].ApiUrl,
+        req.params.date,
+        req.user.token,
+      )
+
+      const results = utilities.processOvertimeShifts(apiShiftsResponse, apiOvertimeShiftsResponse)
+
+      res.render('pages/calendar', {
+        shiftNotifications,
+        tab: 'Calendar',
+        startDate: req.params.date,
+        data: results,
+        uid: req.user.username,
+        employeeName: req.user.employeeName,
+        csrfToken: res.locals.csrfToken,
+        hmppsAuthMFAUser: req.hmppsAuthMFAUser,
+        authUrl: req.authUrl,
+      })
+    } catch (error) {
+      serviceUnavailable(req, res)
+    }
+  }),
+)
+
+// eslint-disable-next-line func-names
+router.get('*', function (req, res) {
+  logger.info('Catch and redirect to current month view')
```

Comment:
Is this redirect useful for users? I feel like it could be hiding a lot of routing errors

Reply:
this whole route needs rewriting. out of scope.